test/advection_test/test_host.F90

```diff
@@ -357,212 +405,480 @@ subroutine test_host(retval, test_suites)
+            errflg_final = -1 ! Notify test script that a failure occured
+         end if
+      else
+         ! Reset error flag to continue testing other properties:
+         errflg = 0
+      end if
+      ! Mass mixing ratio:
       call const_props(index_ice)%is_mass_mixing_ratio(const_log, errflg,     &
            errmsg)
       if (errflg /= 0) then
          write(6, '(a,i0,a,a,i0,/,a)') "ERROR: Error, ", errflg, " trying ",  &
               "to get mass mixing ratio prop for cld_ice index = ",           &
               index_ice, trim(errmsg)
-         errflg_final = -1 !Notify test script that a failure occured
+         errflg_final = -1 ! Notify test script that a failure occured
       end if
       if (errflg == 0) then
          if (.not. const_log) then
             write(6, *) "ERROR: cloud ice is not a mass mixing_ratio"
-            errflg_final = -1 !Notify test script that a failure occured
+            errflg_final = -1 ! Notify test script that a failure occured
          end if
       else
-         !Reset error flag to continue testing other properties:
+         ! Reset error flag to continue testing other properties:
+         errflg = 0
+      end if
+      ! Check mass mixing ratio for a dynamic constituent
+      call const_props(index_dyn2)%is_mass_mixing_ratio(const_log, errflg,     &
+           errmsg)
+      if (errflg /= 0) then
+         write(6, '(a,i0,a,a,i0,/,a)') "ERROR: Error, ", errflg, " trying ",  &
+              "to get mass mixing ratio prop for dyn_const2 index = ",           &
+              index_dyn2, trim(errmsg)
+         errflg_final = -1 ! Notify test script that a failure occured
+      end if
+      if (errflg == 0) then
+         if (.not. const_log) then
+            write(6, *) "ERROR: dyn_const2 is not a mass mixing_ratio"
+            errflg_final = -1 ! Notify test script that a failure occured
+         end if
+      else
+         ! Reset error flag to continue testing other properties:
          errflg = 0
       end if
 
-      !Dry mixing ratio:
+      ! Dry mixing ratio:
       call const_props(index_ice)%is_dry(const_log, errflg, errmsg)
       if (errflg /= 0) then
          write(6, '(a,i0,a,a,i0,/,a)') "ERROR: Error, ", errflg, " trying ",  &
               "to get dry prop for cld_ice index = ", index_ice, trim(errmsg)
-         errflg_final = -1 !Notify test script that a failure occurred
+         errflg_final = -1 ! Notify test script that a failure occurred
       end if
       if (errflg == 0) then
          if (.not. const_log) then
             write(6, *) "ERROR: cloud ice mass_mixing_ratio is not dry"
             errflg_final = -1
          end if
       else
-         !Reset error flag to continue testing other properties:
+         ! Reset error flag to continue testing other properties:
+         errflg = 0
+      end if
+      ! Check moist mixing ratio for a dynamic constituent
+      call const_props(index_dyn2)%is_dry(const_log, errflg, errmsg)
```

Comment:
Should this be calling "is_moist"?

Reply:
added another testing code block to confirm that is_moist returns true (but keeping the test of is_dry() returning false)